docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+    // ...other logic for onSubmit
+  }
+}
+
+
+const styles = StyleSheet.create({
+  view: { backgroundColor: 'white'},
+  submittedView: {borderWidth: 1}
+});
+```
+
+- We are using the `hasSubmitted` flag to represent whether or not we want to apply `styles.submittedView`. If the style was dynamic then it makes sense to store the style object in state
+- `accessibility` is now explicitly passed to the View component as a boolean. This differs from the prior implementation where `accessibility` wasn’t passed as a prop in initial render but in this case we know the non-specification of `accessibility` is handled in the same way as `accessibilty={false}`
+
+Be wary of your assumptions as uncaught subtleties can introduce differences in behavior! It’s a good idea to have snapshot tests of your component as they will highlight any differences pre and post your migration.
+
+#### Move the call to `requireNativeComponent` to a separate file
+
+This will prepare for the JS to be ready for the new codegen system for the new architecture. The new file should be named `<ComponentName>NativeComponent.js.`
+
+```javascript
+// 1. MyNativeView.js
+// Old way
+const RNTMyNativeView = requireNativeComponent('RNTMyNativeView');
+...
+  return <RNTMyNativeView />
+// New way
+import RNTMyNativeViewNativeComponent from 'RNTMyNativeViewNativeComponent';
+...
+  return <RNTMyNativeViewNativeComponent />
+
+// RNTMyNativeViewNativeComponent.js
+import {requireNativeComponent} from 'react-native';
+const RNTMyNativeViewNativeComponent = requireNativeComponent(
+  'RNTMyNativeView',
+);
+export default RNTMyNativeViewNativeComponent;
+```
+
+**[Flow users]** If `requireNativeComponent` is not typed, you can temporarily use the `mixed` type to fix the Flow warning, for example:
+
+```javascript
+import type { HostComponent } from 'react-native/Libraries/Renderer/shims/ReactNativeTypes';
+// ...
+const RCTWebViewNativeComponent: HostComponent<mixed> =
+  requireNativeComponent < mixed > 'RNTMyNativeView';
+```
+
+#### Migrating off `dispatchViewManagerCommand`
+
+Similar to one above, in an effort to avoid calling methods on the UIManager, all view manager methods are now called through an instance of `NativeCommands`. `codegenNativeCommands` is a new API to code-generate `NativeCommands` given an interface of your view manager’s commands.
+
+**Before**
+
+```javascript
+class MyComponent extends React.Component<Props> {
+  _moveToRegion: (region: Region, duration: number) => {
+   **** **UIManager.dispatchViewManagerCommand(
+      ReactNative.findNodeHandle(this),
+      'moveToRegion',
+      [region, duration],
+    );**
+  }
+
+  render() {
+    return <MyCustomMapNativeComponent
+       onPress={this._moveToRegion} />
+  }
+}
+```
+
+**Creating the NativeCommands with `codegenNativeCommands`**
+
+```javascript
+// MyCustomMapNativeComponent.js
+import codegeNativeCommands from 'codegenNativeCommands';
```

Comment:
Is this supposed to be a relative path or some package?

Reply:
Yup this should be:

```suggestion
import codegenNativeCommands from 'react-native/Libraries/Utilities/codegenNativeCommands';
```